components/script/dom/gpudevice.rs

```diff
@@ -641,4 +645,74 @@ impl GPUDeviceMethods for GPUDevice {
 
         GPUCommandEncoder::new(&self.global(), self.channel.clone(), encoder, true)
     }
+    /// https://gpuweb.github.io/gpuweb/#dom-gpudevice-createsampler
+    fn CreateSampler(&self, descriptor: &GPUSamplerDescriptor) -> DomRoot<GPUSampler> {
+        let (sender, receiver) = ipc::channel().unwrap();
+        let sampler_id = self
+            .global()
+            .wgpu_id_hub()
+            .lock()
+            .create_sampler_id(self.device.0.backend());
+        let compare_enable = descriptor.compare.is_some();
+        let desc = SamplerDescriptor {
+            label: Default::default(),
+            address_mode_u: assign_address_mode(descriptor.addressModeU),
+            address_mode_v: assign_address_mode(descriptor.addressModeV),
+            address_mode_w: assign_address_mode(descriptor.addressModeW),
+            mag_filter: assign_filter_mode(descriptor.magFilter),
+            min_filter: assign_filter_mode(descriptor.minFilter),
+            mipmap_filter: assign_filter_mode(descriptor.mipmapFilter),
+            lod_min_clamp: *descriptor.lodMinClamp,
+            lod_max_clamp: *descriptor.lodMaxClamp,
+            compare: if let Some(c) = descriptor.compare {
+                match c {
+                    GPUCompareFunction::Never => CompareFunction::Never,
+                    GPUCompareFunction::Less => CompareFunction::Less,
+                    GPUCompareFunction::Equal => CompareFunction::Equal,
+                    GPUCompareFunction::Less_equal => CompareFunction::LessEqual,
+                    GPUCompareFunction::Greater => CompareFunction::Greater,
+                    GPUCompareFunction::Not_equal => CompareFunction::NotEqual,
+                    GPUCompareFunction::Greater_equal => CompareFunction::GreaterEqual,
+                    GPUCompareFunction::Always => CompareFunction::Always,
+                }
+            } else {
+                CompareFunction::Undefined
+            },
+        };
+        self.channel
+            .0
+            .send(WebGPURequest::CreateSampler {
+                sender,
+                device_id: self.device.0,
+                sampler_id,
+                descriptor: desc,
+            })
+            .expect("Failed to create WebGPU sampler");
+
+        let sampler = receiver.recv().unwrap();
```

Comment:
One of the expected outcomes from our ID management rewrite was to *not* do `recv` here. Resource creation should be fully asynchronous:
  - we create a new ID
  - we ask the GPU side to create a resource and associate it with an ID, by sending a message
  - we return to the script, no need to wait for the GPU process to send us anything back

Reply:
Sorry, I did not realize that. I'll remove the sender and receiver and make a new `WebGPUSampler` object in the script itself.

Reply:
I guess we can refactor this for the existing resources too, without needing to do anything described in https://github.com/servo/servo/issues/25472. I'll open a separate PR for the for those.

Reply:
Yes, that's correct. It's totally fine to do separately, just wanted to point out that this is important.